packages/ts/react-grid/test/autogrid.spec.tsx

```diff
@@ -0,0 +1,73 @@
+import { expect, use } from '@esm-bundle/chai';
+import { GridElement } from '@hilla/react-components/Grid.js';
+import { render } from '@testing-library/react';
+import sinonChai from 'sinon-chai';
+import { AutoGrid } from '../src/autogrid.js';
+import type { CrudService } from '../src/crud.js';
+import Pageable from '../src/types/Pageable.js';
+import { Person, PersonModel } from './TestModels.js';
+//@ts-ignore
+import { getBodyCellContent } from './grid-test-utils.js';
+use(sinonChai);
+
+const fakeService: CrudService<Person> = {
+  list: async (request: Pageable): Promise<Person[]> => {
+    const data: Person[] = [
+      { firstName: 'John', lastName: 'Dove' },
+      { firstName: 'Jane', lastName: 'Love' },
+    ];
+    if (request.pageNumber === 0) {
+      return data;
+    }
+
+    return [];
+  },
+};
+
+async function sleep(ms: number) {
+  return new Promise((resolve) =>
+    setTimeout(() => {
+      resolve(undefined);
+    }, 1),
+  );
+}
+describe('@hilla/react-grid', () => {
+  beforeEach(() => {});
+
+  function TestAutoGrid() {
+    return <AutoGrid service={fakeService} model={PersonModel}></AutoGrid>;
+  }
+  describe('useAutoGrid', () => {
+    it('creates columns based on model', async () => {
+      const result = render(<TestAutoGrid />);
+      const columns = result.container.querySelectorAll('vaadin-grid-sort-column');
+      expect(columns.length).to.equal(2);
+      expect(columns[0].path).to.equal('firstName');
+      expect(columns[0].header).to.equal('First name');
+      expect(columns[1].path).to.equal('lastName');
+      expect(columns[1].header).to.equal('Last name');
+    });
+    it('sets a data provider', async () => {
+      const result = render(<TestAutoGrid />);
+      const grid = result.container.querySelector('vaadin-grid');
+      expect(grid?.dataProvider).to.not.be.undefined;
+    });
+    it('data provider provides data', async () => {
+      const result = render(<TestAutoGrid />);
+      const grid: GridElement = result.container.querySelector('vaadin-grid')!;
+      grid.requestContentUpdate();
+      await sleep(1);
```

Comment:
Interesting that the `flushGrid` helper doesn't work here. Still waiting for the next frame seems to be enough, which might be better than arbitrary timeouts. The `requestContentUpdate` doesn't make a difference and can be removed.

Reply:
This is also fixed in the next PR, let me move the fix here

Reply:
Removed `requestContentUpdate`

Reply:
I tried adding the `nextRender` helper from https://github.com/vaadin/testing-helpers/blob/fb02eb62812de679e1534967ada8dbfd90795a95/src/utils.ts#L67 but depending on `@vaadin/testin-helpers` fail the tests with
```
10:16:45 AM [vite] Internal server error: Failed to resolve import "/__web-dev-server__web-socket.js" from ".vite/deps/@vaadin_testing-helpers.js?v=0c03f7fe". Does the file exist?
```

and copying the helper into the project causes
```
  Uncaught ReferenceError: JSCompiler_renameProperty is not defined
```
because of the Polymer import

Reply:
Using `flushGrid` now, it seems to work

Reply:
What fixes it is the `await`, even though `flushGrid` doesn't return a promise. If you remove the `await` the test fails. I would suggest to wait for an animation frame instead to make this more explicit.

Reply:
which helper would that be and from where?

Reply:
We use a `nextFrame` helper in WCs, which comes from `node_modules/@vaadin/testing-helpers/dist/utils.js`

It's basically this:
```js
/**
 * Resolves after requestAnimationFrame.
 *
 * @return {Promise<void>} Promise resolved after requestAnimationFrame
 */
export async function nextFrame() {
    return new Promise((resolve) => {
        requestAnimationFrame(() => {
            resolve();
        });
    });
}
```

Reply:
Updated